### ISSUE TITLE
`fake_ide` does not report the right error

### ISSUE THREAD

**Issue by maximedenes** [coq/coq#7095]
Steps to reproduce: checkout coq/coq@01b7de3a673eb89cea61442c4db721aad9520c9f, add https://github.com/gares/coq/blob/076efd5f8d17f1b73bcc7669641b89f86819e3c9/test-suite/ide/bug7088.fake to the test-suite, and see that the `Arguments ... : assert` error is not reported. Instead, a message about API limitations is shown.

**Comment by maximedenes** [comment]
See dicussion in #7090.

**Comment by ejgallego** [comment]
In my opinion this is not a problem of fake-ide essentially [even if we could spend cycles fixing it] but more of a consequence of the current API messing with the document + limitations of the actual parsing code. IMHO this is very low priority and should solve by itself at some point.

**Comment by maximedenes** [comment]
@ejgallego can you explain more? Do you mean this problem is specific to the message about the state parsing? Or can `fake_ide` pick up the wrong error in other scenarios?

**Comment by ejgallego** [comment]
The first problem is that current API takes away the control over the document from the client. So imagine you add a new span "foo." on top of the span 30. Then, Coq sometimes will add _two_ spans, like in this case. That means you need to wait for Coq as you cannot tell it to name the new span as you would like. It also means is easy to get confused as to which span you are really in.

Second limitation is that due to "proof mode" problems --- among others --- the STM doesn't synchronize parsing state when requested to parse in any other part of the document than the actual one. This is not gonna be fixed easily and thus this but won't either.

> @ejgallego can you explain more? Do you mean this problem is specific to the message about the state parsing? Or can fake_ide pick up the wrong error in other scenarios?

The error in a sense is correct, `fake_ide` is not at fault here. There are many combinations of calls in the STM API that while should work, they don't as of today. [For example coqtop cannot use `observe` but has to actually join documents as to avoid parsing problems]

**Comment by ejgallego** [comment]
The `fake_ide` log indicates that `fake_ide` is broken I think:

```
Add ((" Qed. ",-9),(10,true))
GOOD (7,(Inr 8,""))
Add ((" Arguments id T x : assert. ",-10),(1,true))
```
So why does `fake_ide` adds at 1 here, that doesn't make sense, it should add at 8.

I misread the log and I thought fake ide was trying to add at the original sid returned after the proof, but instead it adds to a random place. That doesn't look good.

**Comment by gares** [comment]
This is another effect of the bug that is fixed in PR #7090

**Comment by ejgallego** [comment]
Umm, how so? Where does 1 come from?
